### PR TITLE
Add golang to html highlighter

### PR DESCRIPTION
**Pull request by Nimor111**
# What

Adding first homework - syntax highlighter for golang that outputs a coloured html file. 

# Who

Георги Божинов - 81333

# Check

- Do not commit binary files
- Do not commit generated project files

### CODE REVIEW

**Comment by googleson78** [comment]
You should check this out - [Monadic parsing](https://adjoint.fun/books/haskell/monparsing.pdf) .

It's really awesome!

File: homework/81333/golang-highlighter/src/Data/TokenTypes.hs [hunk 1]
```diff
@@ -0,0 +1,289 @@
+{-# LANGUAGE GADTs             #-}
```
Comment (googleson78) [inline homework/81333/golang-highlighter/src/Data/TokenTypes.hs]: Why do you need `GADTs` here, a normal data type does the same job, as far as I can see?

File: homework/81333/golang-highlighter/src/Data/Highlighter.hs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Data.Highlighter
+  ( tokensToHtml
+  ) where
+
+import qualified Data.Text       as T
+import qualified Data.Text.IO    as TI
```
[the remark this answers is not in the document]
Comment (georgi-bozhinov) [reply homework/81333/golang-highlighter/src/Data/Highlighter.hs]: Didn't know that, thanks.

[the remark this answers is not in the document]
Comment (Nimor111) [reply homework/81333/golang-highlighter/src/Data/Highlighter.hs]: Thank you, did not know that.

**Pull request merged**